bindata/v3.11.0/openshift-apiserver/deploy.yaml

```diff
@@ -1,23 +1,32 @@
 apiVersion: apps/v1
-kind: DaemonSet
+kind: Deployment
 metadata:
   namespace: openshift-apiserver
   name: apiserver
   labels:
     app: openshift-apiserver
     apiserver: "true"
 spec:
-  updateStrategy:
-    type: RollingUpdate
+  replicas: 3
```

Comment:
what does this mean for single-node clusters? Will we be always progressing?

Reply:
shouldn't that be observed from the number of masters in the cluster? if someone scales them to 5?

Reply:
3 instances in a 5 master node cluster at least sounds sane and doesn't break.

Reply:
> what does this mean for single-node clusters? Will we be always progressing?

Good catch. The logic should be determining health by checking whether the replica count is equal or greater to the number of masters to maintain the invariant previously guaranteed by the daemonset.

Reply:
> 3 instances in a 5 master node cluster at least sounds sane and doesn't break.

Is that preferable to adjusting the replica count according to the number of masters that are present?

Reply:
single node clusters are not a supported platform.  Refinements could be made, but I don't see them as blocking us stablizing upgrades for supported platforms.